src/librustc/lint/builtin.rs

```diff
@@ -2036,6 +2036,35 @@ impl LintPass for HardwiredLints {
     }
 }
 
+declare_lint! {
+    UNEXPORTED_NO_MANGLE,
```

Comment:
In terms of a lint name, we may want to carefully consider [our conventions](https://github.com/rust-lang/rfcs/blob/cc53afbe5dea41e1f7d1c3dce71e013abe025211/text/0344-conventions-galore.md) in this area. This may not be the best name as `allow(unexported_no_mangle)` may not quite fit those conventions.

Some other possible names may be:

* `private-no-mangle-fns`
* `priv-no-mangle-fns`
* `private-unmangled-fns`

Or some permutation of prefixes/suffixes. Our conventions say that because this only applies to functions it should probably have "fns" in the name somewhere though.